lib/service/local_notification_service.dart

```diff
@@ -0,0 +1,218 @@
+    final NotificationDetails platformChannelSpecifics =
+        NotificationDetails(android: androidPlatformChannelSpecifics);
+    await _localNotificationService.show(
+        generateIntFromString(id), title, body, platformChannelSpecifics,
+        payload: payload);
+  }
+
+  Future<void> groupNotifications() async {
+    AndroidNotificationChannelGroup channelGroup =
+        const AndroidNotificationChannelGroup('com.my.app.alert1', 'mychannel1');
+    await _localNotificationService
+        .resolvePlatformSpecificImplementation<
+            AndroidFlutterLocalNotificationsPlugin>()
+        ?.createNotificationChannelGroup(channelGroup);
+    List<ActiveNotification>? activeNotifications =
+        await _localNotificationService
+            .resolvePlatformSpecificImplementation<
+                AndroidFlutterLocalNotificationsPlugin>()
+            ?.getActiveNotifications();
+
+    if (activeNotifications != null && activeNotifications.isNotEmpty) {
+      List<String> lines =
+          activeNotifications.map((e) => e.title.toString()).toList();
+      InboxStyleInformation inboxStyleInformation = InboxStyleInformation(
+        lines,
+        contentTitle: "${activeNotifications.length - 1} Updates",
+        summaryText: "${activeNotifications.length - 1} Updates",
+      );
+      AndroidNotificationDetails groupNotificationDetails =
+          AndroidNotificationDetails(
+        channelGroup.id,
+        channelGroup.name,
+        channelDescription: channelGroup.description,
+        styleInformation: inboxStyleInformation,
+        setAsGroupSummary: true,
+        groupKey: channelGroup.id,
+        // onlyAlertOnce: true,
+      );
+      NotificationDetails groupNotificationDetailsPlatformSpefics =
+          NotificationDetails(android: groupNotificationDetails);
+      await _localNotificationService.show(
+          0, '', '', groupNotificationDetailsPlatformSpefics);
+    }
+  }
+
+  Future<List<PendingNotificationRequest>> pendingNotificationRequests() async {
+    return await _localNotificationService.pendingNotificationRequests();
+  }
+
+  Future<void> cancelNotificationById(String id) async {
+    await _localNotificationService.cancel(generateIntFromString(id));
+  }
+
+  Future<void> cancelAllNotifications() async {
+    await _localNotificationService.cancelAll();
+  }
+
+  void onDidReceiveLocalNotification(
+      int id, String? title, String? body, String? payload) async {
+    if (payload == null) {
+      return;
+    }
+    final message = message_class.Message.fromJson(
+        jsonDecode(utf8.decode(payload.runes.toList())));
+    onNotificationClick.add(message);
+  }
+
+  Future<String> handleAction(String actionModule, String actionTable) async {
+    final provider = providers[actionModule];
+    if (provider == null) {
+      return "";
+    }
+    final information = provider[actionTable];
+    if (information == null) {
+      return "";
+    }
+    final path = information.item1;
```

Comment:
Why don't you directly return information.item1 ?

Reply:
It was to print result